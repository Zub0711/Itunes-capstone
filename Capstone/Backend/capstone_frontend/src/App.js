import "./App.css";
import Homepage from "./components/Homepage";
import Search from "./components/Searchpage";
import { Route, Routes } from "react-router-dom";
// I start of by importing my components homepage & search.I then import route & routes from react-router-dom

function App() {
  // I then add my routes and return the path to my components
  return (
    <div className="App" data-testid="componentTest">
      <Routes>
        <Route path="/" element={<Homepage />}></Route>
        <Route path="/search" element={<Search />}></Route>
      </Routes>
    </div>
  );
}

export default App;
