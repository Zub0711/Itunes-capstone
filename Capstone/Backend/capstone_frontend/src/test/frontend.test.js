import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../App";

// In this test I will be creating a snapshot test for the app.js component
// I start of by using test and set my test message and create a arrow function
// I then assign the varibale const tree and renderer.create to snapshot test my app.js component
test("snapshot test", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
// I then create a unit text for my app component
// use it and set the message to "component renders"

it("Component renders", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  // I then use screen.getbytestid to get the id of the component
  const component = screen.getByTestId("componentTest");
  // and I use expect and toBeInTheDocument to conduct the test
  expect(component).toBeInTheDocument();
});
