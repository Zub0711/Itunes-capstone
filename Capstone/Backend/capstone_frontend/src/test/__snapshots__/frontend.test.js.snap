// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="App"
  data-testid="componentTest"
>
  <div
    className="body"
  >
    <div
      className="container"
    >
      <nav
        className="bg-body-tertiary navbar navbar-expand-lg navbar-light"
      >
        <div
          className="container"
        >
          <span
            className="navbar-brand"
          >
            üçé Tunes 
          </span>
        </div>
      </nav>
    </div>
    <div
      className="search-path"
    >
      <h1>
        Welcome !!
      </h1>
      <h4>
        To
      </h4>
      <h1>
        üçé Tunes
      </h1>
      <p>
        Welcome to üçé Tunes to proceed to the search page click the button below ‚Üì
      </p>
    </div>
    <div
      className="button-con"
    >
      <a
        href="/search"
        onClick={[Function]}
      >
        <button
          className="btn active btn-primary btn-lg"
          disabled={false}
          type="button"
        >
          Go to Search
        </button>
         
      </a>
    </div>
    <div
      className="Footer"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <blockquote
            className="blockquote mb-0"
          >
            <p>
              üçé Tunes is App that uses a Itunes API to retrieve music
            </p>
            <footer
              className="blockquote-footer"
            >
              <cite
                title="Source Title"
              >
                Created by Zubair Hoosain
              </cite>
            </footer>
          </blockquote>
        </div>
      </div>
    </div>
  </div>
</div>
`;
